src/models/document/document-content.ts

```diff
@@ -307,6 +312,9 @@ export const DocumentContentModel = types
       self.tileMap.delete(tileId);
     },
     moveRowToIndex(rowIndex: number, newRowIndex: number) {
+      if (newRowIndex === 0) {
```

Comment:
This restriction should only be in force for documents that have section headers. I need to think about the best way to accomplish this in the long run. Probably we need to add `isSectioned` as a property on the `DocumentContentModel`.